### PR TITLE
chore: enable REST clients for all working APIs

### PR DESCRIPTION
**Pull request by codyoss**
Changes:

feat(accessapproval): start generating REST clients

feat(accesscontextmanager): start generating REST clients

feat(apigateway): start generating REST clients

feat(appengine): start generating REST clients

feat(artifactregistry): start generating REST clients

feat(asset): start generating REST clients

feat(assuredworkloads): start generating REST clients

feat(automl): start generating REST clients

feat(baremetalsolution): start generating REST clients

feat(batch): start generating REST clients

feat(beyondcorp): start generating REST clients

feat(bigquery): start generating REST clients

feat(billing): start generating REST clients

feat(binaryauthorization): start generating REST clients

feat(certificatemanager): start generating REST clients

feat(channel): start generating REST clients

feat(clouddms): start generating REST clients

feat(cloudtasks): start generating REST clients

feat(contactcenterinsights): start generating REST clients

feat(datafusion): start generating REST clients

feat(dataproc): start generating REST clients

feat(datastore): start generating REST clients

feat(datastream): start generating REST clients

feat(debugger): start generating REST clients

feat(deploy): start generating REST clients

feat(dialogflow): start generating REST clients

feat(dlp): start generating REST clients

feat(documentai): start generating REST clients

feat(essentialcontacts): start generating REST clients

feat(eventarc): start generating REST clients

feat(filestore): start generating REST clients

feat(firestore): start generating REST clients

feat(functions): start generating REST clients

feat(gaming): start generating REST clients

feat(gkebackup): start generating REST clients

feat(gkemulticloud): start generating REST clients

feat(gsuiteaddons): start generating REST clients

feat(iam): start generating REST clients

feat(iap): start generating REST clients

feat(ids): start generating REST clients

feat(iot): start generating REST clients

feat(language): start generating REST clients

feat(memcache): start generating REST clients

feat(metastore): start generating REST clients

feat(monitoring): start generating REST clients

feat(networkmanagement): start generating REST clients

feat(networksecurity): start generating REST clients

feat(optimization): start generating REST clients

feat(orchestration): start generating REST clients

feat(orgpolicy): start generating REST clients

feat(osconfig): start generating REST clients

feat(oslogin): start generating REST clients

feat(policytroubleshooter): start generating REST clients

feat(pubsub): start generating REST clients

feat(recommender): start generating REST clients

feat(redis): start generating REST clients

feat(resourcemanager): start generating REST clients

feat(resourcesettings): start generating REST clients

feat(run): start generating REST clients

feat(scheduler): start generating REST clients

feat(secretmanager): start generating REST clients

feat(security): start generating REST clients

feat(securitycenter): start generating REST clients

feat(servicecontrol): start generating REST clients

feat(servicedirectory): start generating REST clients

feat(servicemanagement): start generating REST clients

feat(serviceusage): start generating REST clients

feat(shell): start generating REST clients

feat(spanner): start generating REST clients

feat(speech): start generating REST clients

feat(storagetransfer): start generating REST clients

feat(talent): start generating REST clients

feat(texttospeech): start generating REST clients

feat(trace): start generating REST clients

feat(translate): start generating REST clients

feat(video): start generating REST clients

feat(videointelligence): start generating REST clients

feat(vision/v2): start generating REST clients

feat(vmmigration): start generating REST clients

feat(vpcaccess): start generating REST clients

feat(webrisk): start generating REST clients

feat(websecurityscanner): start generating REST clients

feat(workflows): start generating REST clients

### CODE REVIEW

**Comment by codyoss** [comment]
The only issue here now is with caused by: https://github.com/googleapis/google-cloud-go/issues/6415. Should be good to merge.

**Review by noahdietz — APPROVED** [review]

**Review by quartzmo — CHANGES_REQUESTED** [review]
Why are the snippets only for aiplatform and artifactregistry? Was the generation process cut short somehow?

File: aiplatform/apiv1beta1/deployment_resource_pool_client.go [hunk 1]
```diff
@@ -0,0 +1,672 @@
...
+		WaitOperation:                []gax.CallOption{},
+	}
+}
+
+// internalDeploymentResourcePoolClient is an interface that defines the methods available from Vertex AI API.
+type internalDeploymentResourcePoolClient interface {
+	Close() error
+	setGoogleClientInfo(...string)
+	Connection() *grpc.ClientConn
+	CreateDeploymentResourcePoolOperation(name string) *CreateDeploymentResourcePoolOperation
```
Comment (quartzmo) [inline aiplatform/apiv1beta1/deployment_resource_pool_client.go:100]: Where is the `CreateDeploymentResourcePool` method in this client? I don't see it in the test either.

I'm trying to find the impl used in this snippet:

```
c, err := aiplatform.NewDeploymentResourcePoolClient(ctx)
	if err != nil {
		// TODO: Handle error.
	}
	defer c.Close()

	req := &aiplatformpb.CreateDeploymentResourcePoolRequest{
		// TODO: Fill request struct fields.
		// See https://pkg.go.dev/google.golang.org/genproto/googleapis/cloud/aiplatform/v1beta1#CreateDeploymentResourcePoolRequest.
	}
	op, err := c.CreateDeploymentResourcePool(ctx, req)
```

File: aiplatform/apiv1beta1/deployment_resource_pool_client.go [hunk 2]
```diff
@@ -0,0 +1,672 @@
...
+	GetOperation(context.Context, *longrunningpb.GetOperationRequest, ...gax.CallOption) (*longrunningpb.Operation, error)
+	ListOperations(context.Context, *longrunningpb.ListOperationsRequest, ...gax.CallOption) *OperationIterator
+	WaitOperation(context.Context, *longrunningpb.WaitOperationRequest, ...gax.CallOption) (*longrunningpb.Operation, error)
+}
+
+// DeploymentResourcePoolClient is a client for interacting with Vertex AI API.
+// Methods, except Close, may be called concurrently. However, fields must not be modified concurrently with method calls.
+//
+// A service that manages the DeploymentResourcePool resource.
+type DeploymentResourcePoolClient struct {
```
Comment (quartzmo) [inline aiplatform/apiv1beta1/deployment_resource_pool_client.go:118]: Is there a reason why this client doesn't have a `restClient`? Something to do with only having LRO RPCs?

File: aiplatform/apiv1beta1/deployment_resource_pool_client.go (hunk 1 above)
Comment (codyoss) [reply aiplatform/apiv1beta1/deployment_resource_pool_client.go:100]: There is none. aiplatform does not yet REST transports. It is likely there is a diff between last regen and this PR that caused a diff of a new feature to be added here. But this is unrelated to transport stuff.

**Comment by codyoss** [comment]
@quartzmo must have forgot to revert the snippets. I will do that. I will let the next regen handle landing all of the new snippets. That takes more memory to compute than my mac can handle well :P

**Comment by noahdietz** [comment]
Need to resolve some questions that came up offline, will follow up with y'all directly.

**Review by quartzmo — COMMENTED** [review]
SGTM

**Comment by codyoss** [comment]
I am going to just close this PR for now. Will re-create once we are sure we can move forward.

**Pull request closed without merging**